### PR TITLE
feat(SegmentationMask): add instance segmentation message

### PR DESCRIPTION
**Pull request by StepTurtle**
## Description

Add a new message type to send instance segmentation results.

## Related links
Parent Issue:
- https://github.com/autowarefoundation/autoware.universe/issues/7235

Discussion:
- https://github.com/orgs/autowarefoundation/discussions/5047

Parent PR:
- https://github.com/autowarefoundation/autoware.universe/pull/8165

## Effects on system behavior

Currently, we plan to use this message type at perception camera pipeline.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by mitsudome-r** [comment]
Note:
This is being discussed in the following discussion thread: https://github.com/orgs/autowarefoundation/discussions/5047

**Comment by StepTurtle** [comment]
@xmfcx can you review?

**Comment by StepTurtle** [comment]
Closing the discussion and PRs as `won't do` due to a lack of motivation, requests on this topic, and no feedback received for a while.

File: msg/SegmentationConfig.msg
Comment (mitsudome-r) [inline msg/SegmentationConfig.msg:1]: If we only need `autoware_perception_msgs/ObjectClassification[] classification` for this message, maybe we can just merge this with the SegmentationMask message. (Of course we can also move the comments as well)

Comment (StepTurtle) [reply msg/SegmentationConfig.msg:1]: Updated as you said.

related commit: https://github.com/autowarefoundation/autoware_internal_msgs/pull/23/commits/8a35861dbfe06e459d19e167efe4a1a6037853d7

**Comment by amadeuszsz** [comment]
@mitsudome-r 
I'm looking on all messages in `autoware_internal_msgs` and I'm not sure if this repository is best place for `SegmentationMask`. Shouldn't we place it in `autoware_perception_msgs`? Sorry, if I missed some comment about architecture/design decision and I'm wrong.

**Comment by amadeuszsz** [comment]
@StepTurtle 
After discussion, we would like to introduce your new msg into [autoware_msgs](https://github.com/autowarefoundation/autoware_msgs). Could you open new PR for that package? After that, please update your PRs which depend on this msg type and close this PR.

**Comment by mitsudome-r** [comment]
Let me talk with @xmfcx on this during Software WG call.
Generally speaking, autoware_msgs is for inter-component message, and we would like to put messages used internally within a component (perception module in this case) to be placed in this repo.

We are planning to refactor this repository similarly to autoware_msgs so that we have autoware_internal_perception_msgs as well.

**Comment by amadeuszsz** [comment]
@mitsudome-r 
Understood. Therefore looking forward for an update after next Software WG meeting.

@StepTurtle 
Sorry for inconvenience regarding requested change of your msg from my side. With reference to @mitsudome-r comment, please give us a bit time and we will back to you with final design decision. I believe this is the last remaining step to merge RTMDet into autoware.universe.

**Comment by xmfcx** [comment]
Sorry for the delays, I will open up a PR in autoware_internal_msgs to solve this problem.

**Pull request closed without merging**